src/stravalib/client.py

```diff
@@ -638,12 +638,78 @@ def get_friend_activities(self, limit: int | None = None) -> NoReturn:
             "See https://developers.strava.com/docs/january-2018-update/"
         )
 
+    def _validate_activity_type(
+        self,
+        params: dict[str, Any],
+        activity_type: str | None,
+        sport_type: str | None,
+    ) -> dict[str, Any]:
+        """Validate activity type and sport type values.
+
+        Parameters
+        ----------
+        params : dict
+            A dictionary of activity values used to update or create an
+            activity.
+        activity_type : str, default=None
+            The activity type (case-insensitive).
+            Deprecated. Prefer to use sport_type. In a request where both type
+            and sport_type are present, this field will be ignored.
+            See https://developers.strava.com/docs/reference/#api-models-UpdatableActivity.
+            For possible values see: :class:`stravalib.model.Activity.TYPES`
+        sport_type : str, default=None
+            For possible values (case-sensitive) see:
+            :class:`stravalib.model.Activity.SPORT_TYPES`
+
+        Returns
+        -------
+        dict
+            Dictionary containing overlapping parameter values between the two
+            methods
+        Raises
+        ------
+        ValueError
+            If the activity_type or sport_type is invalid.
+        """
+
+        # Check for sport_type first. If provided, it is favored over
+        # activity_type / type
+        if sport_type is not None:
+            if not sport_type in model.Activity.SPORT_TYPES:
+                raise ValueError(
+                    f"Invalid activity type: {sport_type}. Possible values: {model.Activity.SPORT_TYPES!r}"
+                )
+            params["sport_type"] = sport_type
```

Comment:
I removed the `pop(type)` here because it wasn't necessary anymore (since you changed the order of checking the types to `sport_type` first).